Files/ViewModels/ItemViewModel.cs

```diff
@@ -806,10 +802,45 @@ public async void RapidAddItemsToCollectionAsync(string path, string previousDir
                 }
                 else
                 {
-                    if (await EnumerateItemsFromStandardFolderAsync(path))
+                    var sourcePageType = AssociatedInstance.ContentFrame.SourcePageType;
+                    if (await EnumerateItemsFromStandardFolderAsync(path, _currentStorageFolder, sourcePageType, _addFilesCTS.Token, cacheOnly: false))
                     {
                         WatchForDirectoryChanges(path);
                     }
+
+                    if (App.AppSettings.UseFileListCache && !_addFilesCTS.IsCancellationRequested)
+                    {
+                        // run background tasks to iterate through folders and cache all of them preemptively
+                        var folders = _filesAndFolders.Where(e => e.PrimaryItemAttribute == StorageItemTypes.Folder);
+                        var currentStorageFolderSnapshot = _currentStorageFolder;
+                        Task.Run(async () =>
+                        {
+                            try
+                            {
+                                await folders.AsyncParallelForEach(async (folder) =>
+                                {
+                                    if (_addFilesCTS.IsCancellationRequested) return;
+
+                                    var path = folder.ItemPath;
+                                    StorageFolderWithPath storageFolder = null;
+                                    if (Path.IsPathRooted(path))
+                                    {
+                                        var res = await FilesystemTasks.Wrap(() => StorageFileExtensions.DangerousGetFolderWithPathFromPathAsync(path, null, parentFolder: currentStorageFolderSnapshot));
+                                        if (res)
+                                        {
+                                            storageFolder = res.Result;
+                                        }
+                                    }
+                                    await EnumerateItemsFromStandardFolderAsync(path, storageFolder, sourcePageType, _addFilesCTS.Token, cacheOnly: true);
+                                }, maxDegreeOfParallelism: 5);
```

Comment:
@gave92 @yaichenbaum I setup 5 as a start, but i believe this should be an option in settings?

Reply:
@jakoss What does this number control?

Reply:
On every navigation i'm running preemptive caching on child folders. This numbers controls the number of folders to cache in parallel. The larger the number - the more disc and cpu usage user can see

Reply:
5 seems like a good start, there should probably be a setting though.

Reply:
I added new setting. PR is ready for review